doc/source/serve/scaling-and-resource-allocation.md

```diff
@@ -0,0 +1,160 @@
+:::
+
+**downscale_delay_s[default_value=600.0]**: The config is to control how long the cluster needs to wait before scaling down replicas.
+
+**upscale_delay_s[default_value=30.0]**: The config is to control how long the cluster need to wait before scaling up replicas.
+
+:::{note}
+`downscale_delay_s` and `upscale_delay_s` are to control the frequency of doing autoscaling work. For example, if your use case takes a long time to do initialization work, you can increase `downscale_delay_s` to make the downscaling happen slowly.
+:::
+
+**smoothing_factor[default_value=1]**: The multiplicative factor to speedup/slowdown each autoscaling step. For example, when the use case has high traffic volume in short period of time, you can increase `smoothing_factor` to scale up the resource quickly.
+
+**metrics_interval_s[default_value=10]**: This is control how often each replica sends metrics to the autoscaler. (Normally you don't need to change this config.)
+
+(serve-cpus-gpus)=
+
+## Resource Management (CPUs, GPUs)
+
+You may want to specify a deployment's resource requirements, for example if a deployment requires a GPU.  To assign hardware resources per replica, you can pass resource requirements to
+`ray_actor_options`.
+By default, each replica requires one CPU.
+To learn about options to pass in, take a look at [Resources with Actors guide](actor-resource-guide).
+
+For example, to create a deployment where each replica uses a single GPU, you can do the
+following:
+
+```python
+@serve.deployment(ray_actor_options={"num_gpus": 1})
+def func(*args):
+    return do_something_with_my_gpu()
+```
+
+(serve-fractional-resources-guide)=
+
+### Fractional Resources
+
+Suppose you have two models and each doesn't fully saturate a GPU.  You might want to have them share a GPU by allocating 0.5 GPUs each.
+
+To do this, the resources specified in `ray_actor_options` can be *fractional*.
+This allows you to flexibly share resources between replicas.
+For example, if you have two models and each doesn't fully saturate a GPU, you might want to have them share a GPU by allocating 0.5 GPUs each.
+
+```python
+@serve.deployment(ray_actor_options={"num_gpus": 0.5})
+def func_1(*args):
+    return do_something_with_my_gpu()
+
+@serve.deployment(ray_actor_options={"num_gpus": 0.5})
+def func_2(*args):
+    return do_something_with_my_gpu()
+```
+
+In this example, each replica of each deployment will be allocated 0.5 GPUs.  The same can be done to multiplex over CPUs, using `"num_cpus"`.
+
+(serve-omp-num-threads)=
+
+## Configuring Parallelism with OMP_NUM_THREADS
+
+Deep learning models like PyTorch and Tensorflow often use multithreading when performing inference.
+The number of CPUs they use is controlled by the OMP_NUM_THREADS environment variable.
+To [avoid contention](omp-num-thread-note), Ray sets `OMP_NUM_THREADS=1` by default because Ray workers and actors use a single CPU by default.
```

Comment:
> To [avoid contention](omp-num-thread-note), Ray sets `OMP_NUM_THREADS=1` by default because Ray workers and actors use a single CPU by default.

What is a Ray worker in this context? Is it a task? Can we use a more precise term here?

Reply:
I would guess task but it could mean a Ray worker process, I'm not sure if there's some subtlety here.  @simon-mo can you confirm?